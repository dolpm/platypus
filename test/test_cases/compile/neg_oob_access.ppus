pipe main [] |> unit {
  mut vector[int] foo = Vector_new <| [];
  loop 0->20 as i {
    Vector_push <| [~&foo, %i];
  }

  int len = Vector_length <| [&foo];
  ~&int last = Vector_get_mut <| [~&foo, len - 1];

  Printnl <| [last];

  /* pop off elements to shrink vecbuf */
  loop 0->18 as i {
    Vector_pop <| [~&foo];
  }

  /* shouldn't point to valid memory -- not possible! */
  Printnl <| [last];

  |> ();
}
