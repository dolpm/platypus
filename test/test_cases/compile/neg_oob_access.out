Fatal error: exception Failure("variable foo can't be borrowed after it has already been mutably borrowed.")
